undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -449,7 +450,11 @@ public Duration getDefaultSessionTimeout() {
             codecConfigurator.configure(support).build(serviceTarget).install();
         }
 
-        infoBuilder.install();
+        try {
+            infoBuilder.install();
+        } catch (DuplicateServiceException e) {
+            throw new DuplicateServiceException("Host and context path are occupied, " + deploymentInfoServiceName + " can't be registered. " + e.getMessage());
```

Comment:
For any exceptions thrown they need to be added via org.wildfly.extension.undertow.logging.UndertowLogger - this way they will also have a unique error ID and can be translated to different languages.

Reply:
Sorry and thanks for that. Hope it will be okay now